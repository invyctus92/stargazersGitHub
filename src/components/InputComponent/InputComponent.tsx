/**
 * Generic input field
 *
 * @format
 */

 import React, { type PropsWithChildren } from "react";
 import { StyleSheet, TextInput } from "react-native";
 
 const InputComponent: React.FC<
   PropsWithChildren<{
     placeholder?: string;
     value: string;
     onChangeText: (value: string) => void;
   }>
 > = ({ placeholder = "", value = "ciao", onChangeText = () => {} }) => {
   return (
     <TextInput
       placeholder={placeholder}
       value={value}
       onChangeText={(value) => {
         onChangeText(value);
       }}
       style={styles.inputContainer}
     />
   );
 };
 
 const styles = StyleSheet.create({
   inputContainer: {
     width: "80%",
     backgroundColor: "#ffffff",
     borderRadius: 15,
     borderWidth: 1,
     borderColor: "#3d3d3d",
     padding: 10,
   },
 });
 
 export default InputComponent;
 