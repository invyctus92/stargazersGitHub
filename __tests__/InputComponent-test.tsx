

import React, { useState } from 'react';
import renderer from 'react-test-renderer';
import InputComponent from '../src/components/InputComponent/InputComponent';

test('renders correctly', () => {
  const tree = renderer.create(<InputComponent value="" onChangeText={() => {}} />).toJSON();
  expect(tree).toMatchSnapshot();
});

import { render, screen, fireEvent } from '@testing-library/react-native';


test('input does not change its text by writing ', () => {
  const value = '';
  const onChangeTextMock = jest.fn();

  render(<InputComponent value={value} onChangeText={onChangeTextMock}  placeholder={"Write the repo"} />);

  const CHANGE_TEXT = 'content';

 
  fireEvent.changeText(screen.getByPlaceholderText("Write the repo"), CHANGE_TEXT);

  expect(onChangeTextMock).toBeCalledWith(CHANGE_TEXT);
  expect(value).toEqual("")
});


