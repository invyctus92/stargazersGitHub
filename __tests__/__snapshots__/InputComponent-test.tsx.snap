// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<TextInput
  onChangeText={[Function]}
  placeholder=""
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#3d3d3d",
      "borderRadius": 15,
      "borderWidth": 1,
      "padding": 10,
      "width": "80%",
    }
  }
  value=""
/>
`;
